RELEASE.md

```diff
@@ -0,0 +1,106 @@
+# Release process
+
+This file documents the necessary steps for releasing Irmin to its various users
+(via GitHub, `opam-repository` and Tezos).
+
+At a high level, releasing Irmin consists of publishing the following artefacts:
+
+- a Git [commit tag][git-tags];
+- a set of documentation on GitHub pages (e.g. [`mirage.github.io/irmin`][pages-docs]);
+- a release archive (`.tbz` file containing the project source) on GitHub;
+- a set of `.opam` files in [`opam-repository`][opam-repo] that point to this
+  release archive;
+- (optionally) a copy of these `.opam` files in the Tezos
+  [`opam-repository`][tezos-opam-repo].
+
+Most of this can be handled automatically by [`dune-release`][dune-release], as
+described in the instructions below.
+
+[git-tags]: https://git-scm.com/book/en/v2/Git-Basics-Tagging
+[pages-docs]: https://mirage.github.io/irmin
+[dune-release]: https://github.com/ocamllabs/dune-release
+[opam-repo]: https://github.com/ocaml/opam-repository
+
+## Releasing to opam-repository and GitHub
+
+- Check that no `.opam` files contain `pin-depends` fields. If so, release those
+  packages first.
+
+```sh
+; git grep -A 10 "pin-depends" *.opam
+```
+
+- Make a pull-request to this repository containing pre-release changes (drop
+  `pin-depends`, add release number to `CHANGES.md`, anu new constraints) and an
+  empty commit to host the release tag.
+
+```sh
+; git fetch upstream
+; git checkout -B release-X.Y.Z upstream/main
+; git commit -m "Prepare X.Y.Z release" -- CHANGES.md
+; git commit --allow-empty -m "Release X.Y.Z"
+; hub pull-request
+```
+
+- Wait for the CI to pass on the release PR, then perform the following steps to
+  release to `opam-repository`:
+
+```sh
+; opam pin add odoc.2.0.0 https://github.com/ocaml/odoc.git  # Use a modern Odoc
+
+; dune-release tag                        #  Create appropriate Git tag by reading CHANGES.md
+; dune-release distrib --skip-tests       #  Build release archive
+; dune-release publish distrib --verbose  #  Push release archive to GitHub
+; dune-release publish doc --verbose      #  Push documentation to GitHub pages
+; dune-release opam pkg                   #  Generate `opam` files for `opam-repository`
+; dune-release opam submit                #  Make PR to `opam-repository`
```

Comment:
what's wrong with:
```
$ dune-release tag  
$ dune-release
```
?

Reply:
I do `dune-release tag; dune-release` as well

Reply:
I use those for smaller repositories too. For Irmin, I find the `--skip-tests` and `--verbose` flags helpful, so tend towards the individual stages here.

Reply:
Maybe this means we need a `dune-release --verbose --skip-tests` option?

Reply:
That's also a good point. I think `dune-release` allows the `--verbose` flag, but not the `skip-tests` one. I've opened [an issue](https://github.com/ocamllabs/dune-release/issues/357) also for that.